pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,194 @@
+		if persistentIPsNetworkName, err := getPersistentIPsNetworkName(virtClient, namespace, network.Multus.NetworkName); err != nil {
+			return map[string]IPAMClaimParams{}, fmt.Errorf("failed retrieving persistentIPsNetworkName: %w", err)
+		} else if persistentIPsNetworkName != "" {
+			networkToIPAMClaimName[network.Name] = IPAMClaimParams{
+				claimName:   fmt.Sprintf("%s.%s", vmiName, network.Name),
+				networkName: persistentIPsNetworkName,
+			}
+		}
+	}
+
+	return networkToIPAMClaimName, nil
+}
+
+func getPersistentIPsNetworkName(virtClient kubecli.KubevirtClient, namespace string, fullNetworkName string) (string, error) {
+	namespace, networkName := getNamespaceAndNetworkName(namespace, fullNetworkName)
+	nad, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.Background(), networkName, v1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to locate network attachment definition %s/%s", namespace, networkName)
+	}
+
+	if nad.Spec.Config == "" {
+		return "", nil
+	}
+
+	netConf := struct {
+		AllowPersistentIPs bool   `json:"allowPersistentIPs,omitempty"`
+		Name               string `json:"name,omitempty"`
+	}{}
+	err = json.Unmarshal([]byte(nad.Spec.Config), &netConf)
+	if err != nil {
+		return "", fmt.Errorf("failed to unmarshal NAD spec.config JSON: %v", err)
+	}
+
+	if !netConf.AllowPersistentIPs {
+		return "", nil
+	}
+
+	if netConf.Name == "" {
+		return "", fmt.Errorf("failed to obtain network name: cannot be empty when allowPersistentIPs is true")
+	}
+
+	return netConf.Name, nil
+}
+
+func CreateIPAMClaims(client kubecli.KubevirtClient, vmi *virtv1.VirtualMachineInstance, ownerRef *v1.OwnerReference) error {
+	nonAbsentNetworks := filterNonAbsentNetworks(vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks)
+	networkToIPAMClaimParams, err := GetNetworkToIPAMClaimParams(client, vmi.Namespace, vmi.Name, nonAbsentNetworks)
+	if err != nil {
+		return fmt.Errorf("failed composing networkToIPAMClaimName: %w", err)
+	}
+
+	// For a standalone VMI, the IPAMClaim owner is the VMI itself
+	if ownerRef == nil {
+		ownerRef = v1.NewControllerRef(vmi, virtv1.VirtualMachineInstanceGroupVersionKind)
+	}
+
+	claims := composeIPAMClaims(vmi.Namespace, ownerRef, networkToIPAMClaimParams)
+	err = createIPAMClaims(client, vmi.Namespace, claims)
+	if err != nil {
+		return fmt.Errorf("failed IPAMClaims creation for VMI %s: %w", vmi.Name, err)
+	}
+
+	return nil
+}
+
+func composeIPAMClaims(namespace string, ownerRef *v1.OwnerReference, networkToIPAMClaimParams map[string]IPAMClaimParams) []*ipamclaims.IPAMClaim {
+	claims := []*ipamclaims.IPAMClaim{}
+	for netName, ipamClaimParams := range networkToIPAMClaimParams {
+		claims = append(claims, composeIPAMClaim(
+			namespace,
+			*ownerRef,
+			ipamClaimParams.claimName,
+			ipamClaimParams.networkName,
+			namescheme.GenerateHashedInterfaceName(netName),
```

Comment:
This should be done inside `IPAMClaim`

Reply:
but then the function need to get netName instead, so it is 1 for 1